### PR TITLE
M3-613 Delete image

### PR DESCRIPTION
**Pull request by Jskobos**
**Purpose**

As a user, I would like to be able to delete an image.

### CODE REVIEW

File: src/features/Images/ImagesLanding.tsx [hunk 1]
```diff
@@ -56,6 +56,11 @@ interface State {
     imageID?: string,
     label?: string,
   };
+  removeDialog: {
+    open: boolean,
```
Comment (rjstires) [inline src/features/Images/ImagesLanding.tsx]: Needs a submitting state.

File: src/features/Images/ImagesLanding.tsx [hunk 2]
```diff
@@ -106,19 +114,35 @@ class ImagesLanding extends React.Component<CombinedProps, State> {
...
   }
 
+  removeImage = () => {
+    if (!this.state.removeDialog.imageID) { return; }
+    deleteImage(this.state.removeDialog.imageID)
+      .then(() => {
+        this.closeRemoveDialog();
+      })
+      .catch((error) => {
+        console.log(error);
```
Comment (rjstires) [inline src/features/Images/ImagesLanding.tsx]: console.log left behind.

File: src/features/Images/ImagesLanding.tsx [hunk 3]
```diff
@@ -130,6 +154,18 @@ class ImagesLanding extends React.Component<CombinedProps, State> {
     )
   }
 
+  openRemoveDialog = (image: string, imageID: string) => {
+    this.setState({
+      removeDialog: { open: true, image, imageID },
+    });
+  }
+
+  closeRemoveDialog = () => {
```
Comment (rjstires) [inline src/features/Images/ImagesLanding.tsx]: We need to establish a pattern for delaying the resetting of values (other than open) in this kind of dialogs. The user briefly sees "undefined" as the dialog is being dismissed. I've successfully done it with a timeout, but it's gross.

Comment (Jskobos) [reply src/features/Images/ImagesLanding.tsx]: Do we need to reset the other values to undefined when closing? They are always set to new values before the dialog is remounted, so there is no chance that stale values will be shown to the user.

Comment (rjstires) [reply src/features/Images/ImagesLanding.tsx]: Yea that'll work.

Comment (asauber) [reply src/features/Images/ImagesLanding.tsx]: I personally think the latter is fine.

**Pull request merged**